PR for #88: Add latex template for slides

ew487 opened the pull request:
Closes #88 

In this issue, we added a LaTeX file with a basic beamer shell for slides to [/source/talk] in which we enable the option `\setbeameroption{hide notes}` by default. We added to the LaTeX builder a handout option similar to the one we have for LyX
under the hood, which switches `hide notes` to `show notes`.

ew487 commented:
@jmshapir I reverted the implementation to use the `note` command!

Do you want to use the same kind of input control as we did for the lyx handout?
![image](https://github.com/JMSLab/Template/assets/108754119/e7be9268-c1c2-4e25-9fba-f48aff9365cf)
I think this is causing some of the tests to fail...

jmshapir commented on source/talk/TemplateSlides.tex:
```diff
@@ -0,0 +1,28 @@
+	\end{itemize}
+ 	\item Point Two
+\end{itemize}
+\end{frame}
```
@ew487 can we add an example transition slide between talk sections roughly along the lines of the below?

(Even better, but not essential, if these can be linked to the use of a `\section{}` designation.}

![image](https://github.com/JMSLab/Template/assets/7840353/5abb5157-beb0-45e0-a258-83ba91308bb4)

ew487 replied:
Added [here](https://github.com/JMSLab/Template/pull/89/commits/57caa68034b01a423174e09db52d50ccc68c7d8e)!

jmshapir commented on source/lib/JMSLab/builders/build_latex.py:
Should we add unit testing for new functionality?

ew487 replied:
Added in [daed0b7](https://github.com/JMSLab/Template/pull/89/commits/daed0b70b92865b7d2fa82e46833310207c00c23) and [39b7682](https://github.com/JMSLab/Template/pull/89/commits/39b7682b122a0a99aa48d664640200c951f60dd4)!

jmshapir commented on source/lib/JMSLab/builders/build_latex.py:
```diff
@@ -47,7 +50,83 @@ def add_call_args(self):
-     
+    
+    
+    def check_handout(self, target, env):
```
@ew487 on the below, I guess maintaining parallelism with LyX is ideal if we can do it. If you see a reason why this isn't practical for the TeX builder let me know and we can figure out how best to compromise.

Thanks!

> Do you want to use the same kind of input control as we did for the lyx handout? ![image](https://private-user-images.githubusercontent.com/108754119/290924352-e7be9268-c1c2-4e25-9fba-f48aff9365cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI2NzIwMjMsIm5iZiI6MTcwMjY3MTcyMywicGF0aCI6Ii8xMDg3NTQxMTkvMjkwOTI0MzUyLWU3YmU5MjY4LWMxYzItNGUyNS05ZmJhLWY0OGFmZjkzNjVjZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxNVQyMDIyMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZjU4YWY0MDQzOThiOGFjMmQxZTA1MjA4NzdjNGE4ZTdkNjU3MWE3MjU4OWRmMmY1ODg3MzM3OGQ2MWJlY2ZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BHRMpeHL5ShUwPH7OaQIiMW1QpYfAJL0nb58TT8001c) I think this is causing some of the tests to fail...

ew487 replied:
@jmshapir I think it would work practically! But I just wanted to check because from looking at the unit testing it seems like we're altering the expected behavior of the builder, so if someone based their project off of this template in the past, it might not work anymore?

jmshapir replied:
@ew487 thanks! To clarify, do you mean that the change we're considering here to the LaTeX builder will not be backwards compatible, in the sense that if someone adopts the new builder in an existing repository, they may need to update the syntax they use to call the builder?

If so, that sounds very likely fine. To help me think it through, can you give me an example of syntax that would work with the current builder (i.e. the one in `main`) but not in the alternative we're considering (here in #89)?

jmshapir replied:
To follow up here, while I think it's likely fine to change the syntax of the LaTeX builder, we probably _don't_ want to deprecate any functionality (e.g. bibtex etc.) Anyway we can figure it out once I have a better sense of the issues here thanks.

ew487 replied:
We have not deprecated any functionality and are just requiring that the user enter target files using a specific syntax. So whereas before they are able to write `target = ["Template_tex.pdf", "Template_tex.log"], now we are enforcing that the second target file (if it exists) must be the path to the handout.

Per meeting, @jmshapir is fine with this so I will resolve the thread.

jmshapir reviewed (changes requested):
@ew487 thanks!

In the pull threads I replied to your question and added a few other points.

@rcalvo12 I'm adding you as reviewer both to review code and to recompile the repo. (Locally or on the cluster, as you prefer.)

Thanks!

rcalvo12 commented on source/lib/JMSLab/tests/test_build_latex.py:
```diff
@@ -13,17 +13,19 @@
 
 from ..builders.build_latex import build_latex, LatexBuilder
 from .._exception_classes import ExecCallError
+from .. import misc
```
@ew487 Per spyder, this package is imported but not used. Might want to double check.

rcalvo12 replied:
@ew487 Done in https://github.com/JMSLab/Template/pull/89/commits/11a1c76e2a5668238b3bdb2fd1abd0e10d892a49. Tests ran as usual after the removal so I am resolving.

ew487 replied:
@rcalvo12 Great, thank you!

rcalvo12 reviewed (changes requested):
@ew487 Awesome work on this! Especially the way you cleaned up the latex builder a bit. Just a couple of minor points below.

@jmshapir I also recompiled the repository in 3ed0ae6.

jmshapir commented on output/talk/TemplateSlidesLatex.pdf:
@rcalvo12 thanks!

When we experimented before we were [having an issue with importing EPS files](https://github.com/JMSLab/Template/issues/88#issuecomment-1857786364). It seemed to be specific to the compiler, so if it's working for you, it would be great to add a slide with [this plot](https://github.com/JMSLab/Template/blob/380e12f4d8cb94165fd9f86262894780b008be1e/output/analysis/plots/gdp_educ.eps) for illustration, thanks!

rcalvo12 replied:
@ew487 @jmshapir in https://github.com/JMSLab/Template/pull/89/commits/a2be61ab86ccacb923a2dbb9d1c62a19f0eb6cd1 I went ahead and added plots to both slide templates. I didn't run into any issues compiling with scons locally and all the tests passed, so I think it worked. @ew487 would you be able to confirm?

ew487 replied:
@rcalvo12 Yes, it worked for me too! Thanks!

ew487 commented:
@jmshapir @rcalvo12 Thanks for your comments--I think I've addressed all of them now! If there is anything else, please let me know.

rcalvo12 reviewed (approved):
@ew487 Thanks, barring the reminder posted below this looks good to me!

rcalvo12 commented on output/talk/TemplateSlidesLatex.pdf:
@ew487 Reminder to .gitignore this pdf!

ew487 commented:
@rcalvo12 Thanks for catching that! Added to the .gitignore in [c61131e](https://github.com/JMSLab/Template/pull/89/commits/c61131e035e993fac1b125cf0e6d3919cf3da4b8) and removed in [361a934](https://github.com/JMSLab/Template/pull/89/commits/361a934601158cd07fe9580faac97fe5d55fb65f)

ew487 commented:
Summary [here](https://github.com/JMSLab/Template/issues/88#issuecomment-1893883837)
